tests/e2e/tests/lib.rs

```diff
@@ -1084,3 +1084,133 @@ async fn create_hundred_registrations() {
+        .expect("Failed to mint tokens")
+        .transaction
+        .id;
+    println!("Minted {} cNIGHT. Tx: {}", amount, hex::encode(tx_id));
+
+    // FIXME: it returns first utxo, find by native token or return all utxos
+    let cnight_utxo = match cardano_client
+        .find_utxo_by_tx_id(&cardano_client.address_as_bech32(), hex::encode(tx_id))
+        .await
+    {
+        Some(cnight_utxo) => cnight_utxo,
+        None => panic!("No cNIGHT UTXO found after minting"),
+    };
+
+    let prefix = b"asset_create";
+    let nonce =
+        MidnightClient::calculate_nonce(prefix, cnight_utxo.transaction.id, cnight_utxo.index);
+    println!("Calculated nonce for cNIGHT UTXO: {}", nonce);
+
+    let utxos = cardano_client.utxos().await;
+    assert!(!utxos.is_empty(), "No UTXOs found for funding address");
+    let utxo = utxos
+        .iter()
+        .max_by_key(|u| u.value.lovelace)
+        .expect("No UTXO with lovelace found");
+
+    let deregister_tx = cardano_client
+        .deregister(utxo, &register_tx, &collateral_utxo)
+        .await
+        .expect("Failed to deregister")
+        .transaction
+        .id;
+    println!(
+        "Deregistration transaction submitted with hash: {}",
+        hex::encode(deregister_tx)
+    );
+
+    let reward_address = cardano_client.reward_address_bytes();
+    let dust_address: Vec<u8> = hex::decode(&dust_hex)
+        .expect("Failed to decode DUST hex")
+        .try_into()
+        .unwrap();
+    let events = midnight_client
+        .subscribe_to_cnight_observation_events(&deregister_tx)
+        .await
+        .expect("Failed to listen to cNgD registration event");
+
+    let deregistration = events
+        .iter()
+        .filter_map(|e| e.ok())
+        .filter_map(|evt| evt.as_event::<Deregistration>().ok().flatten())
+        .find(|reg| {
+            reg.0.cardano_reward_address.0 == reward_address
+                && reg.0.dust_public_key.0.0 == dust_address
+        });
+    assert!(
+        deregistration.is_some(),
+        "Did not find deregistration event with expected reward_address and dust_address"
+    );
+    println!(
+        "Matching Deregistration event found: {:?}",
+        deregistration.unwrap()
+    );
+
+    let mapping_removed = events
+        .iter()
+        .filter_map(|e| e.ok())
+        .filter_map(|evt| {
+            evt.as_event::<c_night_observation::events::MappingRemoved>()
+                .ok()
+                .flatten()
+        })
+        .find(|map| {
+            map.0.cardano_reward_address.0 == reward_address
+                && map.0.dust_public_key.0.0 == dust_bytes
+                && map.0.utxo_tx_hash.0 == register_tx_id
+        });
+    assert!(
+        mapping_removed.is_some(),
+        "Did not find MappingRemoved event with expected reward_address, dust_address, and utxo_id"
+    );
+    println!(
+        "Matching MappingRemoved event found: {:?}",
+        mapping_removed.unwrap()
+    );
```

Comment:
We can remove that part, other tests are verifying it. This test's goal is to check if dust is produced after user deregistration -> please add assertions from `cnight_produces_dust` test.